server/cli/src/main/java/org/elasticsearch/cli/MultiCommand.java

```diff
@@ -74,4 +75,16 @@ protected void execute(Terminal terminal, OptionSet options) throws Exception {
         }
         subcommand.mainWithoutErrorHandling(Arrays.copyOfRange(args, 1, args.length), terminal);
     }
+
+    @Override
+    public void close() throws IOException {
+        if (subcommands.isEmpty()) {
+            throw new IllegalStateException("No subcommands configured");
+        }
+        for (Command command : subcommands.values()) {
```

Comment:
Please use `IOUtils.close()` instead of iterating over the subcommands, as this will ensure all subcommands are closed, even on failures.

Reply:
I forgot cli classes were moved out to their own jar. We definitely shouldn't add lucene core just to get IOUtils. But we do need to mimic the behavior here, otherwise one command failing to close could cause the others not to close.